File: levelledmobs-plugin-bukkit/src/main/java/me/lokka30/levelledmobs/bukkit/listeners/ListenerHandler.java
package me.lokka30.levelledmobs.bukkit.listeners;

import java.util.Set;
import me.lokka30.levelledmobs.bukkit.utils.Log;
import org.jetbrains.annotations.NotNull;

public final class ListenerHandler {

    /* vars */

    private final Set<ListenerWrapper> listeners = Set.of(
        new EntitySpawnListener(),
        new PlayerJoinListener()
    );

    /* methods */

    public boolean load() {
        Log.inf("Registering listeners.");
        for(var listener : getListeners())
            if(!listener.register())
                return false;
        return true;
    }

    /* getters and setters */

    @NotNull
    public Set<ListenerWrapper> getListeners() { return listeners; }

}

File: levelledmobs-plugin-bukkit/src/main/kotlin/me/lokka30/levelledmobs/bukkit/listeners/ListenerHandler.kt
package me.lokka30.levelledmobs.bukkit.listeners

import me.lokka30.levelledmobs.bukkit.LevelledMobs.Companion.logInf

/*
FIXME comment
 */
class ListenerHandler {

    /*
    FIXME comment
     */
    val listeners = setOf(
        EntitySpawnListener(),
        PlayerJoinListener()
    )

    /*
    FIXME comment
     */
    fun load() {
        logInf("Registering listeners...")
        listeners.forEach(ListenerWrapper::register)
    }
}